src/inputs/labeled-inputs.scss

```diff
@@ -28,169 +40,163 @@
-      margin-right: $iui-m;
-    }
+    > .iui-input-group {
+      display: flex;
+      gap: $iui-m;
 
-    > .iui-message {
-      margin: 0;
+      > .iui-checkbox,
+      > .iui-radio,
+      > .iui-toggle-switch {
+        margin-right: $iui-m;
 
-      svg:not(.iui-icon) {
-        margin-right: 0;
+        @supports (gap: $iui-m) {
+          margin-right: 0;
+        }
+      }
+
+      &:not(:last-child) {
+        margin-right: $iui-m;
       }
     }
 
     > .iui-input,
     > .iui-textarea {
       padding-right: $iui-icons-default + $iui-l;
 
-      + .iui-message {
-        position: absolute;
-        top: 50%;
-        right: 0;
-        transform: translateY(-50%);
-
-        svg:not(.iui-icon) {
-          margin-right: $iui-sm;
-        }
-      }
-
       &:last-child {
         padding-right: $iui-sm;
       }
     }
   }
+}
 
-  // #region Statuses
-  &.iui-positive {
-    @include iui-inputs-status(
-      iui-color-foreground-positive,
-      iui-color-foreground-positive-overlay,
-      iui-color-foreground-positive-rgb
-    );
+/// Text label for an input.
+/// Supports .iui-required modifier to show red asterisk.
+@mixin iui-input-label($displayStyle: default) {
+  font-weight: $iui-font-weight-semibold;
+  grid-area: label;
+  align-self: center;
+
+  &.iui-required {
+    &::after {
+      content: ' *';
+      @include themed {
+        color: t(iui-color-foreground-negative);
+      }
+    }
   }
+}
 
-  &.iui-warning {
-    @include iui-inputs-status(
-      iui-color-foreground-warning,
-      iui-color-foreground-warning-overlay,
-      iui-color-foreground-warning-rgb
-    );
+/// Modifier on iui-input-label to place it inline.
+@mixin iui-input-label-inline {
+  margin: 0 $iui-m 0 0;
+
+  &.iui-required {
+    margin-right: $iui-xs * 1.5;
   }
+}
 
-  &.iui-negative {
-    @include iui-inputs-status(
-      iui-color-foreground-negative,
-      iui-color-foreground-negative-overlay,
-      iui-color-foreground-negative-rgb
-    );
+/// Message shown below input
+@mixin iui-input-message {
+  font-size: $iui-font-size-small;
+  grid-area: message;
+  @include themed {
+    color: t(iui-text-color-muted);
+  }
+
+  a {
+    user-select: none;
   }
-  // #endregion
 }
 
-@mixin iui-inputs-status($foregroundColor, $foregroundOverlay, $foregroundRgb) {
+/// Icon to be shown near an input.
+@mixin iui-input-icon {
+  display: flex;
+  grid-area: icon;
+  width: $iui-icons-default;
+  height: $iui-icons-default;
+  align-self: center;
+  @include themed {
+    fill: t(iui-icons-color);
+  }
+
+  &:not(:last-child) {
+    margin-right: $iui-xs;
+  }
+}
+
+/// Modifier on iui-input-icon to place it at the end of input.
+@mixin iui-input-icon-inline {
+  // stylelint-disable -- fix position in IE
+  -ms-grid-row: 3;
+  -ms-grid-column: 2;
+  -ms-grid-column-align: end;
+  // stylelint-enable
+
+  grid-area: inputs;
+  justify-self: end;
+  margin-right: $iui-sm;
+  position: relative;
+
+  &.iui-button {
```

Comment:
Would be nice to have demo for this one, to verify

Reply:
a, ok. There is one.

Reply:
Need to handle other sizes too, I guess...
![image](https://user-images.githubusercontent.com/81580355/132198165-1453b35e-7c48-4e3f-b163-7d63cce83d61.png)

If applying iui-small class on button too, need to arrange spacing..
![image](https://user-images.githubusercontent.com/81580355/132198253-b49c1a69-4263-4393-beb5-3dd9b428b75d.png)